### ISSUE TITLE
8.1.2 triggers a false positive semi rule violation?

### ISSUE THREAD

**Issue by jsg2021** [babel/babel-eslint#567]
See this gist:
https://gist.github.com/jsg2021/171b419b831d0e8a69c7bda22c7036fd

**Comment by jsg2021** [comment]
If I reorder the export keywords it fixes the semi error, but I was under the impression that you would not be able to decorate an export statement eg: 
```js
@decoration
export default class Foobar {}
```
instead we are supposed to use it like:
```js
export default
@decoration
class Foobar {}
```

**Comment by jsg2021** [comment]
interestingly, eslint@4.13.1 does not have a false positive semi rule violation under babel-eslint@8.1.2

**Comment by mysticatea** [comment]
This seems like a bug in Babylon. In the second case, Babylon generates AST as `ClassExpression`. The cause is [this condition](https://github.com/babel/babel/blob/64dfb7574e924be3ad737c33cbc81d619dbc14c6/packages/babylon/src/parser/statement.js#L1369).

**Comment by mysticatea** [comment]
Please track https://github.com/babel/babel/issues/7123.

**Comment by hzoo** [comment]
Will land https://github.com/babel/babel/issues/7123 soon

**Comment by jsg2021** [comment]
@hzoo Did this land yet? I'm still observing it in  8.2.2.

**Comment by jsg2021** [comment]
it’s fixed in the latest of each project. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
